cli/command/system/prune_progress.go

```diff
@@ -0,0 +1,175 @@
+package system
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/progress"
+	"github.com/docker/docker/pkg/streamformatter"
+)
+
+type pruneProgressChan struct {
+	isEventsListening chan bool
+	errorChan         chan error
+}
+
+func runPruneProgress(dockerCli command.Cli, progressChans *pruneProgressChan, options pruneOptions) {
+	eventChan := make(chan string, 1)
+	go monitorRemovingEvents(dockerCli, eventChan, progressChans)
+	objCount, err := countObjectsToRemove(dockerCli, options)
+	if err != nil {
+		progressChans.errorChan <- err
+		return
+	}
+	go outputPruningProgress(dockerCli, eventChan, objCount, progressChans)
+}
+
+func outputPruningProgress(dockerCli command.Cli, events <-chan string, numEvents int, progressStruct *pruneProgressChan) {
+	pipeReader, pipeWriter := io.Pipe()
+	progressOut := streamformatter.NewJSONProgressOutput(pipeWriter, false)
+
+	go func() {
+		var i int64 = 1
+		for s := range events {
+			prog := progress.Progress{
+				ID:         "Progress",
+				Action:     s,
+				Current:    i,
+				Total:      int64(numEvents),
```

Comment:
I need to check to see what happens if this receives more than `numEvents`,  which is possible in this case.